up42/storage.py

```diff
@@ -63,34 +68,42 @@ def _query_paginated(
             response_json = self.auth._request(
                 request_type="GET", url=url + f"&page={page}"
             )
-            results_list += response_json["data"]["content"]
+            if "data" in response_json:
+                response_json = response_json["data"]
+            results_list += response_json["content"]
         return results_list[:limit]
 
     def get_assets(
         self,
-        return_json: bool = False,
+        created_after: Optional[str] = None,
+        created_before: Optional[str] = None,
+        workspace_id: Optional[str] = None,
         limit: Optional[int] = None,
-        sortby: str = "createdAt",
+        sortby: str = "created",
         descending: bool = True,
+        return_json: bool = False,
     ) -> Union[List[Asset], dict]:
         """
-        Gets all assets in the workspace as Asset objects or json.
+        Gets all assets in all your accessible workspaces as Asset objects or json.
 
         Args:
-            return_json: If set to True, returns json object.
+            created_after: Only assets that are created strictly after the provided timestamp, e.g. "2020-01-01",
+            created_before: Only assets that are created strictly before the provided timestamp, e.g. "2020-01-30"
+            workspace_id: Only assets that belong to the provided workspace. Use `storage.workspace_id` to limit to
+                your own workspace.
             limit: Optional, only return n first assets by sorting criteria and order.
                 Optimal to select if your workspace contains many assets.
-            sortby: The sorting criteria, one of "createdAt", "updatedAt", "source", "type",
-                "name", "size".
+            sortby: The sorting criteria, one of "created", "updated", "workspaceId", "name", "size".
             descending: The sorting order, True for descending (default), False for ascending.
+            return_json: If set to True, returns json object.
 
         Returns:
             Asset objects in the workspace or alternatively json info of the assets.
         """
         allowed_sorting_criteria = [
```

Comment:
also we have sort by producerName and collectionName

Reply:
Added new sorting parameters to docstring, removed code limitation.